lib/service/configuration_service.dart

```diff
@@ -839,11 +842,27 @@ class ConfigurationServiceImpl implements ConfigurationService {
   }
 
   @override
-  List<PlayListModel>? getPlayList() {
+  Future<List<PlayListModel>?> getPlayList() async {
     final playListsString = _preferences.getStringList(PLAYLISTS) ?? [];
-    return playListsString
+    final playlists = playListsString
         .map((e) => PlayListModel.fromJson(jsonDecode(e)))
         .toList();
+    final hiddenTokens = getTempStorageHiddenTokenIDs();
+
+    final hiddenAddresses =
+        await injector.get<AccountService>().getHiddenAddressIndexes();
```

Comment:
The `ConfigurationService` should only contain the logic to get/set the preferences info. Should not involve any other services or database execution.

if the logic is complex, please move out to another service. Eg `PlaylistService` that contains `ConfigurationService`, `AccountService`, `TokenDao` then free to add your logic there.